File: server/game/logic/Components.lua
local MakeComponent = require("entitas").MakeComponent

local M = {
    CommandCreate = MakeComponent("CommandCreate","id","data"),
    CommandRemove = MakeComponent("CommandRemove","id"),
    CommandMove = MakeComponent("CommandMove","id","data"),
    CommandUpdate = MakeComponent("CommandUpdate","delta"),
    InputCreateFood = MakeComponent("InputCreateFood","count"),

    BaseData = MakeComponent("BaseData","id","name","spriteid"),
    Position = MakeComponent("Position","x","y"),
    Direction = MakeComponent("Direction","value"),
    Speed = MakeComponent("Speed","value"),
    Size = MakeComponent("Size","value"),
    Color = MakeComponent("Color","r","g","b"),
    Mover = MakeComponent("Mover"),
    Food = MakeComponent("Food"),
    Radius =  MakeComponent("Radius","value"),
    EnterView =  MakeComponent("EnterView","ids"),
    LeaveView =  MakeComponent("LeaveView","ids"),
    Dead =  MakeComponent("Dead"),
    Eat =  MakeComponent("Eat","weight")
}

return M

File: BallAction/Assets/Scripts/Components.cs
using UnityEngine;

namespace Component
{

    [System.Serializable]
    public class BaseData
    {
        public int id;
        public string name;
        public int spriteid;
    }


    [System.Serializable]
    public class Position
    {
        public float x;
        public float y;
    }

    [System.Serializable]
    public class Direction
    {
        public float x;
        public float y;
    }

    [System.Serializable]
    public class Speed
    {
        public float value;
    }

    [System.Serializable]
    public class Color
    {
        public int r;
        public int g;
        public int b;
    }

    [System.Serializable]
    public class Radius
    {
        public float value;
    }

    [System.Serializable]
    public class EnterView
    {

    }

    [System.Serializable]
    public class LeaveView
    {

    }
}